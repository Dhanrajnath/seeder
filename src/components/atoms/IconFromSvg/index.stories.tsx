import { Meta, StoryObj } from "@storybook/react";
import IconFromSvg, { IIconFromSvgProps } from ".";
import SeederSvg from "../../../../public/assets/seeder.svg";
import CalendarSvg from "../../../../public/assets/calendar.svg";
import DocumentDownloadSvg from "../../../../public/assets/document-download.svg";
import PercentageSquareSvg from "../../../../public/assets/percentage-square.svg";

export default {
    title: 'Atoms/IconFromSVG',
    component: IconFromSvg,
    argTypes: {
        onClick: { action: 'clicked' },
    },
} as Meta;

type IconFromSvgType = StoryObj<IIconFromSvgProps>;


export const SeederIcon: IconFromSvgType = {
    args: {
        alt: 'Seeder Logo',
        path: SeederSvg
    }
};

export const CalendarIcon: IconFromSvgType = {
    args: {
        alt: 'calendar',
        path: CalendarSvg
    }
};

export const DocumentDownloadIcon: IconFromSvgType = {
    args: {
        alt: 'document-download',
        path: DocumentDownloadSvg
    }
};

export const PercentageSquareIcon: IconFromSvgType = {
    args: {
        alt: 'percentage-square',
        path: PercentageSquareSvg
    }
};