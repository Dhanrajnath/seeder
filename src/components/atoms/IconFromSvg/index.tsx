export interface IIconFromSvgProps {
    path: string;
    alt: string;
    width?: number;
    height?: number;
};

const IconFromSvg = ({
    path,
    alt,
    width,
    height,
}: IIconFromSvgProps) => {
    return (
        <img
            src={path}
            alt={alt}
            width={width ? width : 'inherit'}
            height={height ? height : 'inherit'}
        />
    );
};

export default IconFromSvg;