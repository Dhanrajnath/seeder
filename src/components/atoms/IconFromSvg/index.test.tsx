import { render } from "@testing-library/react";
import IconFromSvg from ".";
import SeederSvg from "../../../../public/assets/seeder.svg";

describe('IconFromSvg tests', () => {
    it('renders a svg img from path', () => {
        const { getByAltText } = render(<IconFromSvg path={SeederSvg} alt={"Seeder"} />);
        const seederIcon = getByAltText('Seeder');
        expect(seederIcon).toBeInTheDocument();
        expect(seederIcon).toMatchSnapshot();
    });
});