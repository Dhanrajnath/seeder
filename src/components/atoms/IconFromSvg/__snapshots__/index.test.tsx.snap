// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconFromSvg tests renders a svg img from path 1`] = `
<img
  alt="Seeder"
  height="inherit"
  src="seeder.svg"
  width="inherit"
/>
`;
